### PR TITLE
Fix call stack positions

### LINKED ISSUE #11055: Fix stack positions
Kinda broke since "pretty errors"; working on it...

When using `throw` in macros, stack positions (or even the whole stack, which is pretty useless without positions..) are lost.
This is a regression caused by #10863

### PR DESCRIPTION
**Pull request by kLabz**
Fixes #11055 (see test, it's mostly about `throw`ing from macros) although there was a more complex (involving `throw` in a build macro added via `Compiler.addGlobalMetadata` on modules loaded via `Compiler.include`) stack regression that this does fix but I couldn't reduce it enough to make a test out of it. It's a very rare edge case, but still would have been nice to have a test..

This PR keeps `ctx.error` mutable but it is only ever meant to be mutated from `setup_common_context`. Rebinding error reporting should now be done by rebinding `ctx.located_error`.

Note that this removes `For function argument 'v'` lines from "have" / "want" unification errors. Couldn't easily bring it back an I'm not sure it was really useful anyway.

### CODE REVIEW

File: src/core/error.ml [hunk 1]
```diff
@@ -146,6 +151,8 @@ module BetterErrors = struct
 				add_access FunctionReturn;
 			| Invariant_parameter i ->
 				add_access (TypeParameter i);
+			| Unify_custom _ ->
+				add_extra err
```
Comment (Simn) [inline src/core/error.ml:155]: What is this actually doing? It seems to mirror the handling of `acc_messages` exactly, so I don't see what difference this additional field makes.

File: src/typing/callUnification.ml [hunk 2]
```diff
@@ -18,7 +18,13 @@ let rec unify_call_args ctx el args r callp inline force_inline in_overload =
...
 	in
 	let arg_error ul name opt p =
-		let err = Stack [(ul,p); (Custom ("For " ^ (if opt then "optional " else "") ^ "function argument '" ^ name ^ "'"), p)] in
+		let msg = ("For " ^ (if opt then "optional " else "") ^ "function argument '" ^ name ^ "'") in
+		let err = match ul with
+			| Stack s -> Stack (s @ [(Custom msg,p)])
+			| Unify l -> Unify (l @ [(Unify_custom msg)])
+			| Custom parent -> Custom (parent ^ "\n" ^ msg)
+			| _ -> Stack [(ul,p); (Custom (compl_msg msg), p)]
+		in
```
Comment (Simn) [inline src/typing/callUnification.ml:27]: This looks awful... We really need a better data structure. But if this makes things work for the time being we can do it.

File: std/haxe/macro/Expr.hx [hunk 3]
```diff
@@ -1011,12 +1011,18 @@ class Error extends Exception {
 	**/
 	public var pos:Position;
 
+	/**
+		Child error messages, if any.
+	**/
+	public var childErrors:Array<Error>;
```
Comment (Simn) [inline std/haxe/macro/Expr.hx]: Should we really make this public API? That means we can't change it later, and I'm not 100% convinced that this is the data structure we ultimately want to go for.

Comment (kLabz) [reply std/haxe/macro/Expr.hx]: We can make it private and use `@:privateAccess` in our tests if you prefer

File: src/core/error.ml (hunk 1 above)
Comment (kLabz) [reply src/core/error.ml:155]: `acc_messages` are used to determine the "main error" and stuff like that. Keeping it separate avoids some strange mixup but I agree I don't like it much...

Comment (Simn) [reply src/core/error.ml:155]: Ah right, that actually makes a difference then. But shouldn't it appear in the haxe/want version as well in that case?

File: src/typing/callUnification.ml (hunk 2 above)
Comment (kLabz) [reply src/typing/callUnification.ml:27]: Yeah.. I am planning more work and refactoring around error handling after 4.3 (error record, move actual message reporting to its own file, make pretty errors work for interp even in edge cases, stuff like that), I can see if I can improve that then

File: src/core/error.ml (hunk 1 above)
Comment (kLabz) [reply src/core/error.ml:155]: I tried to add it to `Some access_next` below  but uh it got ignored for some reason
